### ISSUE TITLE
Cannot update repair schedule after scheduled repair

### ISSUE THREAD

**Issue by thxph** [thelastpickle/cassandra-reaper#1201]
We are seeing these exceptions in reaper's log after each scheduled repairs. The next scheduled repairs still run normally but it seems the number of segments cannot be updated due to this (we are using adaptive schedule).

```
i.c.s.RepairRunner - Repair amount done 52.0
i.c.s.RepairRunner - Repairs for repair run #50ab2950-dd41-11ec-8ed3-f33fb25095c7 done
i.c.s.RepairRunner - extendedSegments = 0, total segments = 52, percent extended = 0.0
i.c.s.RepairRunner - Percent extended segments: 0.0
i.c.s.CassandraStorage - failed to quorum update repair schedule 7c953b90-dbf8-11ec-8ed3-f33fb25095c7
java.lang.InterruptedException: null
        at com.google.common.util.concurrent.AbstractFuture.get(AbstractFuture.java:485)
        at com.google.common.util.concurrent.AbstractFuture$TrustedFuture.get(AbstractFuture.java:85)
        at io.cassandrareaper.storage.CassandraStorage.updateRepairSchedule(CassandraStorage.java:1483)
        at io.cassandrareaper.service.RepairRunner.reduceSegmentsPerNodeToScheduleForUnit(RepairRunner.java:345)
        at io.cassandrareaper.service.RepairRunner.tuneAdaptiveRepair(RepairRunner.java:329)
        at io.cassandrareaper.service.RepairRunner.maybeAdaptRepairSchedule(RepairRunner.java:314)
        at io.cassandrareaper.service.RepairRunner.endRepairRun(RepairRunner.java:287)
        at io.cassandrareaper.service.RepairRunner.startNextSegment(RepairRunner.java:443)
        at io.cassandrareaper.service.RepairRunner.run(RepairRunner.java:204)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
```

┆Issue is synchronized with this [Jira Task](https://k8ssandra.atlassian.net/browse/K8SSAND-1543) by [Unito](https://www.unito.io)
┆friendlyId: K8SSAND-1543
┆priority: Medium

**Comment by adejanovski** [comment]
Hi @thxph,

I saw the same thing as you did and am currently working on a fix.
I think the schedules are updated since we send async queries, but the thread gets interrupted before we can get the ack from Cassandra, which generates that exception.

Thanks for reporting this.

**Comment by alextechnology** [comment]
Would it be possible to also keep the schedule persistent and not update after a "Run Now" invocation of the schedule?  I've noticed when I run a scheduled repair, it recomputes the scheduled datetime using the configured offset based on the time you invoked "Run Now".  Would be better if the schedule stayed constant to allow for ad-hoc repair runs in between

**Comment by adejanovski** [comment]
Hi @alextechnology, would you mind opening a separate issue for this?
